File: app/src/main/java/pl/mosenko/sunnypodlaskie/util/WeatherDataDetailPresentationModelConverter.java
package pl.mosenko.sunnypodlaskie.util;

import pl.mosenko.sunnypodlaskie.mvp.weatherdatadetail.WeatherDataDetailPresentationModel;
import pl.mosenko.sunnypodlaskie.persistence.entities.WeatherDataEntity;

/**
 * Created by syk on 07.06.17.
 */

public class WeatherDataDetailPresentationModelConverter {
    private WeatherDataDetailPresentationModelConverter() {
    }

    public static WeatherDataDetailPresentationModel convert(WeatherDataEntity weatherDataEntity) {
        WeatherDataDetailPresentationModel presentationModel = new WeatherDataDetailPresentationModel();
        presentationModel.setTitleDetails(WeatherDataUtil.getDetailsTitle(weatherDataEntity.getCity(), weatherDataEntity.getReceivingTime()));
        presentationModel.setTemperature(WeatherDataUtil.getFormattedTemperature(weatherDataEntity.getTemperature()));
        presentationModel.setDescription(weatherDataEntity.getWeatherCondition().getDescription());
        presentationModel.setGetIconResource(WeatherDataUtil.getWeatherIconResourceByCode(weatherDataEntity.getIconKey()));
        presentationModel.setPressure(WeatherDataUtil.getFormattedPressure(weatherDataEntity.getPressure()));
        presentationModel.setWindDetails(WeatherDataUtil.getFormattedWindDetails(weatherDataEntity.getWindSpeed(), weatherDataEntity.getWindDegree()));
        presentationModel.setHumidity(WeatherDataUtil.getFormattedHumidity(weatherDataEntity.getHumidity()));
        presentationModel.setSunrise(WeatherDataUtil.getFormattedTime(weatherDataEntity.getSunrise()));
        presentationModel.setSunset(WeatherDataUtil.getFormattedTime(weatherDataEntity.getSunset()));
        return presentationModel;
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/util/WeatherDataDetailPresentationModelConverter.kt
package pl.mosenko.sunnypodlaskie.util

import pl.mosenko.sunnypodlaskie.mvp.weatherdatadetail.WeatherDataDetailPresentationModel
import pl.mosenko.sunnypodlaskie.persistence.entities.WeatherDataEntity

/**
 * Created by syk on 07.06.17.
 */
object WeatherDataDetailPresentationModelConverter {

    fun convert(weatherDataEntity: WeatherDataEntity): WeatherDataDetailPresentationModel {
        val presentationModel = WeatherDataDetailPresentationModel()
        presentationModel.titleDetails = WeatherDataUtil.getDetailsTitle(weatherDataEntity.city, weatherDataEntity.receivingTime)
        presentationModel.temperature = WeatherDataUtil.getFormattedTemperature(weatherDataEntity.temperature)
        presentationModel.description = weatherDataEntity.weatherCondition?.description
        presentationModel.iconResource = WeatherDataUtil.getWeatherIconResourceByCode(weatherDataEntity.iconKey)
        presentationModel.pressure = WeatherDataUtil.getFormattedPressure(weatherDataEntity.pressure)
        presentationModel.windDetails = WeatherDataUtil.getFormattedWindDetails(weatherDataEntity.windSpeed, weatherDataEntity.windDegree)
        presentationModel.humidity = WeatherDataUtil.getFormattedHumidity(weatherDataEntity.humidity)
        presentationModel.sunrise = WeatherDataUtil.getFormattedTime(weatherDataEntity.sunrise)
        presentationModel.sunset = WeatherDataUtil.getFormattedTime(weatherDataEntity.sunset)
        return presentationModel
    }
}
